server/controllers/oauth.js

```diff
@@ -0,0 +1,45 @@
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+const generator = require('generate-password');
+
+// @route GET /login/register/user
+// @desc Login if user exists else Register user
+// @access Private
+exports.loginRegisterUser = asyncHandler(async (req, res, next) => {
```

Comment:
It's not really common practice to bundle these two methods together. We can just have a login route and a register route instead of doing logic here for both.